esp-hal/MIGRATING-0.22.md

```diff
@@ -235,19 +235,15 @@ https://github.com/rust-embedded/embedded-hal/blob/master/docs/migrating-from-0.
 
 ## Driver constructors now take a configuration and are fallible
 
-The old `new_with_config` constructor have been removed, and `new` constructors now always take
+The old `new_with_config` constructors have been removed, and `new` constructors now always take
 a configuration structure. They have also been updated to return a `ConfigError` if the configuration
 is not compatible with the hardware.
 
 ```diff
 -let mut spi = Spi::new_with_config(
 +let mut spi = Spi::new(
      peripherals.SPI2,
-     Config {
-         frequency: 100.kHz(),
-         mode: SpiMode::_0,
-         ..Config::default()
-     },
+     Config::default().with_frequency(100.kHz()).with_mode(Mode::_0),
```

Comment:
Perhaps neither of these versions are right, since the Config itself has changed between the versions. We can maybe `// ...` out this line, indicating that this paragraph isn't about the configuration itself, but the fallibility of the constructor.

Reply:
As mentioned in my other comment, I felt like the diff markers do a good enough job of drawing attention to the parts which the surrounding text is referring to changing.

I felt like standardizing on making the unrelated steps (not highlighted by any diff markers) match the post-migration state made the most sense (if someone spots something in those areas which doesn't match what they expected, they can search for it in the rest of the guide instead of thinking it hasn't actually changed).

If this line is commented then it's not valid in either version either. Since `new_with_config` took a `Config` as does `new` now. So I am not sure how to interpret your suggestion.

Reply:
Or maybe a simple solution is just to mention that configuration structs have also changed and cross reference the other section? (And leave the code in the "new" state outside of the diffed areas.)

Reply:
Perhaps we can paper over this by saying `/* peripheral configuration */`, but this code doesn't need to compile or anything, just needs to highlight the change. If you feel like the current state is good enough, I won't force a change.

Reply:
How about this: We just pass a `config` without showing the code for creating it. It would look like this:

```diff
-let mut spi = Spi::new_with_config(
+let mut spi = Spi::new(
     peripherals.SPI2,
     config,
-);
+)
+.unwrap();
```

Now it's valid before and after and doesn't draw any attention to how the config was constructed or what it contains.

Let me know if you like this and I'll make the change.

Reply:
Sounds good, thanks!

Reply:
I've made the change and re-worded the commit message.